src/pages/index.js

```diff
@@ -1,140 +1,237 @@
-profileEditValidator.enableValidation();
-addCardValidator.enableValidation();
-
-// Event Handlers
-function fillEditProfileForm() {
-  profileTitleInput.value = profileTitle.textContent;
-  profileDescriptionInput.value = profileDescription.textContent;
+function handleSubmit(
+  request,
+  popupInstance,
+  reset,
+  loadingText = "Saving..."
+) {
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+      if (reset) {
+        popupInstance.reset();
+      }
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
 }
 
-// Event Listeners
+function handleEditProfileSubmit(inputValues) {
+  function makeRequest() {
+    return api.updateUserInfo(inputValues).then((userData) => {
+      userInfo.setUserInfo(userData);
+    });
+  }
+  handleSubmit(makeRequest, profileEditModal, true);
+}
+
+function handleAddCardSubmit(inputValues) {
+  function makeRequest() {
+    return api.createCard(inputValues).then((res) => {
+      renderCard(res);
+    });
+  }
+  handleSubmit(makeRequest, addCardPopup, true);
+}
+
+function handleDeleteButton(card) {
+  deleteImageModal.open();
+  return deleteImageModal.handleDelete(() => {
+    function makeRequest() {
+      return api.deleteCard(card.id).then(() => {
+        card.deleteCard();
+      });
+    }
+    handleSubmit(makeRequest, deleteImageModal, false);
+  });
+}
+
+function handleLikeButton(card) {
+  if (card.isLiked) {
+    api
+      .deleteLike(card.id)
+      .then(() => {
+        card.handleIsLiked(false);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card.id)
+      .then(() => {
+        card.handleIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+}
+
+function handleChangeProfilePicture(inputUrl) {
+  function makeRequest() {
+    return api.updateProfileImage(inputUrl).then((res) => {
+      userInfo.setProfileImage(res);
+    });
+  }
+  handleSubmit(makeRequest, profileImageModal, true);
+}
+
+//ProfileEditButtonEventListeners
 profileEditButton.addEventListener("click", () => {
-  profileEditValidator.hideInputError(profileEditForm);
-  fillEditProfileForm();
-  profileEditPopup.open();
+  const { name, about } = userInfo.getUserInfo();
+  profileTitleInput.value = name;
+  profileDescriptionInput.value = about;
+  profileEditModal.open();
+  formValidators["profile-form"].resetValidation();
 });
-
+//Add Card Button Event Listeners
 addNewCardButton.addEventListener("click", () => {
   addCardPopup.open();
+  formValidators["add-card-form"].resetValidation();
 });
 
-profileEditForm.addEventListener("submit", (e) => {
-  e.preventDefault();
-  profileEditPopup._handleFormSubmit(profileEditForm);
+profileImageEditButton.addEventListener("click", () => {
+  profileImageModal.open();
```

Comment:
* The avatar submit button is active after the 1st submitting. Please, call `resetValidation` to disable it as well